typing/typecore.ml

```diff
@@ -460,6 +460,25 @@ let enter_orpat_variables loc env  p1_vs p2_vs =
   unify_vars p1_vs p2_vs
 
 let rec build_as_type env p =
+  let as_ty = build_as_type_aux env p in
+  match
+    List.find_opt (function
+      | Tpat_constraint _, _, _ -> true
+      | _ -> false
+    ) p.pat_extra
+  with
```

Comment:
I think that the code would be more robust, and in fact easier to read, with a recursive loop on all `pat_extra` things on this pattern, using a non-fragile pattern matching.

Reply:
Will do.

Reply:
Done.
And I also added a test to ensure that none of the constraints are ignored (which indeed wasn't the case before your comment).